sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -556,6 +556,98 @@ Parameters:
 
 |===
 
+==== Dynamic Command Groups
+
+[source,c++]
+----
+namespace ext::oneapi::experimental{
+class dynamic_command_group {
+public:
+  dynamic_command_group(
+      command_graph<graph_state::modifiable> graph, std::vector<std::function<void(handler &)>> cgfList);
+
+  void set_active_cgf(int cgfIndex);
+};
+----
+
+Dynamic command-groups can be added as nodes to a graph. They provide a mechanism that
```

Comment:
Wondering whether to a new node type enum for dynamic command-groups, or are we expecting `node:get_type()` to be a kernel? Which will be true now, but maybe not in the future.

Reply:
I think this makes sense, even if it's just for future proofing the extension.

Reply:
I don't think we'd be able to swap a kernel CG for say a copy CG in any backend would we?

Reply:
Yeah that's true, I was thinking if we could use dynamic command-groups for copy CGs then you could pass a list only copy CG type commands. That's getting pretty speculative, so happy to leave it out for now.

Reply:
I'm not sure if there a use-case for having a dynamic command-group node type. Will the users ever need to know that information?

If not, maybe it's better to let dynamic command-groups have the same type as their underlying cgf. So if it's a kernel cgf it would be a kernel node. If it's a copy cgf it would be a copy node.